core/src/main/kotlin/org/evomaster/core/output/service/TestSuiteWriter.kt

```diff
@@ -1071,4 +1096,47 @@ class TestSuiteWriter {
             .toList()
     }
 
+    private fun getDtos(): List<DtoClass> {
+        val restSampler = sampler as AbstractRestSampler
+        val result = mutableListOf<DtoClass>()
+        restSampler.getActionDefinitions().forEach { action ->
+            action.getViewOfChildren().forEach { child ->
+                if (child is BodyParam) {
+                    val primaryGene = child.primaryGene()
+                    // TODO: Payloads could also be json arrays, analyze ArrayGene
+                    if (primaryGene is ObjectGene) {
+                        // TODO: Determine strategy for objects that are not defined as a component and do not have a name
+                        val dtoClass = DtoClass(primaryGene.refType?:primaryGene.hashCode().toString())
+                        primaryGene.fixedFields.forEach { field ->
+                            try {
+                                dtoClass.addField(getDtoField(field))
+                            } catch (e: Exception) {
+                                // do nothing for the moment
+                            }
+                        }
+                        result.add(dtoClass)
+                    }
+                }
+            }
+        }
+        return result
+    }
+
+    private fun getDtoField(field: Gene): DtoField {
+        val wrappedGene = GeneUtils.getWrappedValueGene(field)
+        return DtoField(field.name, when (wrappedGene) {
+            // TODO: handle nested arrays, objects and extend type system for dto fields
+            is StringGene -> "String"
+            is IntegerGene -> "Integer"
+            is LongGene -> "Long"
+            is DoubleGene -> "Double"
+            is FloatGene -> "Float"
+            is Base64StringGene -> "String"
+            is DateGene -> "String"
+            is TimeGene -> "String"
```

Comment:
you might want to use special types for this, and, in the DTOs, throw an `IllegalArgumentExcpetion` if invalid string. however, for now can leave it as acomment (as we still create invalid strings outside robustness testing, and that is not fixed yet)

Reply:
Maybe it's something the Gene could answer by itself, something like: `gene.getDtoType()` and have that answer back. That would also save us from these giant ifs

One other question I have is, for these specific cases of Time and Date, should we in the future handle them with jodatime objects or similar? Or just set them as strings and keep it simple? For a start I think strings are fine, not sure what you had in mind for the future